### PR TITLE
freac: add missing codecs

### PR DESCRIPTION
**Pull request by Janosch**
#### Description
freac: Added missing audio codecs and DSP libraries
smooth: Removed gtk dependency as it is not needed for macOS

Note: Both changes have been suggested by fre:ac's author.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.1
Xcode 12.3 Build version 12C33

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: audio/freac/Portfile [hunk 1]
```diff
@@ -25,6 +25,21 @@ checksums           rmd160  2f1f9ea871793cec1a8fe02a96fb9c03b1fa0c49 \
 depends_lib         port:boca \
                     port:smooth
 
+depends_run         port:lame \
+                    port:ffmpeg \
```
Comment (mf2k) [inline audio/freac/Portfile:29]: Can `ffmpeg-devel` also be used? If so, use this specifications to allow it:

`path:lib/libavcodec.dylib:ffmpeg`

File: audio/freac/Portfile [hunk 2]
```diff
@@ -25,6 +25,21 @@ checksums           rmd160  2f1f9ea871793cec1a8fe02a96fb9c03b1fa0c49 \
...
+                    port:speex \
+                    port:libsndfile \
+                    port:flac \
+                    port:libopus \
+                    port:wavpack \
+                    port:libfdk-aac \
+                    port:mpg123 \
+                    port:mp4v2 \
+                    port:libsamplerate \
+                    port:rubberband
```
Comment (mf2k) [inline audio/freac/Portfile:41]: Please sort these alphabetically.

**Review by mf2k — CHANGES_REQUESTED** [review]

**Comment by kencu** [comment]
freac will need to be rev-bumped I believe, so the new Portfile is picked up by current MacPorts' installations (if any).

It's odd that these are only run-time deps. Does this software not need to see them during the build, to know where to find them, how to configure, etc?

Usually things like this are "lib" deps, as they are needed at both build time and run time.

**Comment by Janosch** [comment]
@kencu Those dependencies are all optional. The software builds and runs without them but is of limited use as there is only the Core Audio encoder available. freac tries to find those dependencies on runtime and loads them if available.

As macports has no optional/recommended dependencies, I also thought about adding a +minimal variant without those runtime dependencies. If you think that makes sense I can add it. 

I just would like to have the default install as close as possible to the upstream version, therefore I added all those encoder/decoders as runtime depedencies.

File: audio/freac/Portfile (hunk 1 above)
Comment (Janosch) [reply audio/freac/Portfile:29]: Yes. I tested it and ffmpeg-devel works as well.

File: audio/freac/Portfile (hunk 2 above)
Comment (Janosch) [reply audio/freac/Portfile]: Done

**Comment by kencu** [comment]
Thanks for clarifying.

**Review by mf2k — APPROVED** [review]

**Comment by kencu** [comment]
Frank, does this need a revbump? No installed files are changing, but I recall Josh mentioning cached  portfiles need refreshing if things like the runtime deps change.

**Comment by mf2k** [comment]
@kencu : Yes, freac should have the `revision` increased by one.

**Comment by mf2k** [comment]
And it is not needed for smooth.

**Comment by Janosch** [comment]
Increased revision to 1

**Pull request merged**